pkg/data/data_storage.go

```diff
@@ -0,0 +1,50 @@
+package data
+
+import (
+	"time"
+
+	. "github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const collection = "things"
+
+type DataStore struct {
+	Database *mgo.Database
+}
+
+type IDataStore interface {
+	Get()
+	Create()
+}
+
+func NewDataStore(database *mgo.Database) DataStore {
+	return DataStore{database}
+}
```

Comment:
In order to avoid allocations, as a rule of thumb, [always return a pointer](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type) for constructors :eyes: 

```suggestion
func NewDataStore(database *mgo.Database) *DataStore {
	return &DataStore{database}
}
```

Reply:
Done!